sceptre/plan/actions.py

```diff
@@ -1000,3 +1000,109 @@ def diff(self, stack_differ):
         :rtype: sceptre.diffing.stack_differ.StackDiff
         """
         return stack_differ.diff(self)
+
+    @add_stack_hooks
+    def detect_stack_drift(self) -> Tuple[str, Union[str, dict]]:
+        """
+        Detects stack drift for a running stack.
+
+        :returns: The stack drift.
+        """
+        response = self._detect_stack_drift()
+
+        detection_id = response["StackDriftDetectionId"]
+        status = self._wait_for_drift_status(detection_id)
+
+        if status in ["DETECTION_COMPLETE", "DETECTION_FAILED"]:
+            response = self._describe_stack_resource_drifts()
+            return_value = (self.stack.external_name, response)
+        else:
```

Comment:
What's the else-case here? This sort of vague `else` leaves reviewers and maintainers trying to guess what was in your mind. In cases where you're checking against distinct circumstances, I find it much better (leaving no question) to employ something like:

```python
if first_condition:
    ...
elif second_condition:
    ...
else: # Raise an error in the case that you don't expect to be possible because you've already covered all options
    raise SomeError()
```

This sort of approach, while more verbose, is clear to the reader what conditions are being considered.

Reply:
Yes, agreed.

Reply:
Fixed.